lib/datapacksutils.js

```diff
@@ -3042,8 +3042,115 @@ DataPacksUtils.prototype.loadSalesforceMetadata = async function() {
 
 var folderOnlyMetadata = [ "aura", "layouts", "lwc", "staticresources" ]
 
-DataPacksUtils.prototype.retrieveSalesforce = async function(jobInfo, listOnly) {
+DataPacksUtils.prototype.getAdditionalSFMetadataDependencies = async function(jobInfo, sfFolderPath){
+    let vlocityDataPackTypes = [],
+        salesforceKeys = [],
+        salesforceTypes = [];
+
+    vlocityDataPackTypes = Object.keys(await this.vlocity.utilityservice.getVlocityDataPackConfigurations());
+    
+    salesforceKeys = jobInfo.specificManifestKeys.filter(item => {
+        return vlocityDataPackTypes.indexOf(item.split('/')[0]) === -1;
+    });
+
+    salesforceTypes = Array.from(new Set(
+        salesforceKeys.map(item => {
+            return item.split('/')[0];
+        })
+    ));
+
+    if (salesforceTypes.indexOf('classes') !== 0) {
```

Comment:
It would be good to make this more generic so it can work for other types beside ApexClass if we want to add them. This whole part with Apex classes here and when doing the MetadataComponentDependency check can probably be made into a function that accepts type inputs for everything.